frb_codegen/src/utils.rs

```diff
@@ -89,10 +89,59 @@ impl Display for BlockIndex {
     }
 }
 
-#[extend::ext]
-impl std::path::Path {
+pub trait PathExt {
+    fn file_name_str(&self) -> Option<&str>;
+    fn directory_name_str(&self) -> Option<&str>;
+}
+
+impl PathExt for std::path::Path {
     #[inline]
     fn file_name_str(&self) -> Option<&str> {
-        self.file_name().and_then(OsStr::to_str)
+        self.file_name().and_then(std::ffi::OsStr::to_str)
     }
+    #[inline]
+    fn directory_name_str(&self) -> Option<&str> {
+        self.parent().and_then(|p| p.to_str())
+    }
+}
+
+/// given 2 paths, output the relative path from the 1st one to the 2ed one.
+/// # Examples
+/// ```
+/// use lib_flutter_rust_bridge_codegen::utils::relative_path;
+/// let output_1 = relative_path("./a/b/c", "./a/d/e");
+/// assert_eq!(std::path::PathBuf ::from(output_1), std::path::PathBuf ::from("../../d/e"));
+/// let output_2 = relative_path("./a/d/e", "./a/b/c");
+/// assert_eq!(std::path::PathBuf ::from(output_2), std::path::PathBuf ::from("../../b/c"));
+/// let output_3 = relative_path("./a/b/c", "./a/b/c/d");
+/// assert_eq!(std::path::PathBuf ::from(output_3), std::path::PathBuf ::from("d"));
+/// let output_4 = relative_path("./a/b/c", "./a/b/c/");
+/// assert_eq!(std::path::PathBuf::from(output_4),std::path::PathBuf::from(""));
+/// ```
+pub fn relative_path(from: &str, to: &str) -> String {
```

Comment:
would be great to have some `unit tests` for this utils.rs's functions (if we really want to manually write down this)

Reply:
Actually, the doc example here:
```rust
/// # Examples
/// ```
/// use lib_flutter_rust_bridge_codegen::utils::relative_path;
/// let output_1 = relative_path("./a/b/c", "./a/d/e");
/// assert_eq!(std::path::PathBuf ::from(output_1), std::path::PathBuf ::from("../../d/e"));
/// let output_2 = relative_path("./a/d/e", "./a/b/c");
/// assert_eq!(std::path::PathBuf ::from(output_2), std::path::PathBuf ::from("../../b/c"));
/// let output_3 = relative_path("./a/b/c", "./a/b/c/d");
/// assert_eq!(std::path::PathBuf ::from(output_3), std::path::PathBuf ::from("d"));
/// let output_4 = relative_path("./a/b/c", "./a/b/c/");
/// assert_eq!(std::path::PathBuf::from(output_4),std::path::PathBuf::from(""));
/// ```
```
is already unit tests,  rust compiler is so restricted that the compilation would not be passed if just omitting `use lib_flutter_rust_bridge_codegen::utils::relative_path;` here.